todo-list/src/TasksContext.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import TodoItem from "./TodoItem";
+
+const TaskContext = React.createContext()
+
+export class TasksContextProvider extends React.Component {
+    constructor() {
+        super()
+        this.state = {
+            'tasks' : [],
+            'lastId': 0
+        }
+    }
+
+    add = (task) => {
+        if (!task)
+        {
+            return
+        }
+        const newTask = {
+            'id': this.state.lastId + 1,
+            'task': task,
+            'completed': false
+        }
+        this.state.tasks.push(newTask)
+        this.setState(prevState => {
+            return {
+                lastId : prevState.lastId + 1
+            }
+        })
```

Comment:
идентично, но меньше строк
```suggestion
        this.setState({lastId : this.state.lastId + 1});
```